R/civis_ml_utils.R

```diff
@@ -19,19 +19,111 @@ CIVIS_ML_CLASSIFIERS <- c("sparse_logistic",
                           "multilayer_perceptron_classifier",
                           "stacking_classifier")
 
-CIVIS_ML_TEMPLATE_IDS <- data.frame(
-  id = c(9112, 9113, 9968, 9969, 10582, 10583, 11219, 11220, 11221),
-  version = c(1.1, 1.1, 2.0, 2.0, 2.1, 2.1, 2.2, 2.2, 2.2),
-  name = c("train", "predict", "train", "predict", "train", "predict", "train", "predict", "register"),
-  stringsAsFactors = FALSE
-)
+get_train_template_id <- function() {
+
+  civis_ml_template_ids <- get_template_ids_all_versions()
+
+  id <- civis_ml_template_ids[civis_ml_template_ids$version == "prod" &
+                              civis_ml_template_ids$name == "training", "id"]
+
+  return(id)
+
+}
+
+
+#' Get template IDs for all accessible CivisML versions.
+#'
+#' @return  A data frame containing the template id, CivisML version, and job type.
+get_template_ids_all_versions <- function(){
```

Comment:
The function `get_template_ids_all_versions` is called at various places downstream (e.g., inside `get_train_template_id` and `get_predict_template_id`). Is there a convenient way to cache whatever `get_template_ids_all_versions` returns, something similar to [`lru_cache` used in civis-python](https://github.com/civisanalytics/civis-python/blob/e5787ed84d7a78904ca9d9400b070a6ac2bd275d/civis/ml/_model.py#L299)? It'd be nice to avoid making multiple API calls when calling `get_template_ids_all_versions` multiple times in the same runtime session, since the results should be the same.

Reply:
Yep.

Adding the following should work:
```
get_template_ids_all_versions <- memoise::memoise(get_template_ids_all_versions)
```